Does not build with recent TGUI

TGUI 0.9 has introduced some breaking changes, our code will need to be adapted to them.

```
build/obj/game/boe.menu.cpp:200:52: error: cannot convert ‘const sf::String’ to ‘const tgui::String&’
  200 |         return this->tgui.get<tgui::MenuBar>(this->internal_menubar_widget_name);
      |                                              ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                    |
      |                                                    const sf::String
```

```
build/obj/game/boe.menu.cpp:238:30: error: ‘using element_type = class tgui::MenuBar’ {aka ‘class tgui::MenuBar’} has no member named ‘disconnect’
  238 |                 if(!menubar->disconnect(connection_id))
      |                              ^~~~~~~~~~
```

and possibly more.

--- a/src/game/boe.menu.cpp
+++ b/src/game/boe.menu.cpp
@@ -235,7 +235,7 @@ void OpenBoEMenu::update_for_game_state(eGameMode overall_mode, bool party_in_me
 // Disconnect all spell menu items from signals and clear the menus
 void OpenBoEMenu::purge_spell_menus(tgui::MenuBar::Ptr& menubar) {
 	for(const auto& connection_id : this->spell_menus_connection_ids) {
-		if(!menubar->disconnect(connection_id))
+		if(!menubar->onMenuItemClick.disconnect(connection_id))
 			throw std::runtime_error { "BUG: attempted to disconnect menubar signal using invalid connection id" };
 	}
 

--- a/src/game/boe.menu.hpp
+++ b/src/game/boe.menu.hpp
@@ -26,11 +26,11 @@ public:
 
 private:
 
-	using MenuHierarchy = std::vector<sf::String>;
+	using MenuHierarchy = std::vector<tgui::String>;
 
 	tgui::Gui tgui;
 	cUniverse& univ;
-	const sf::String internal_menubar_widget_name { "openboe-menu" };
+	const tgui::String internal_menubar_widget_name { "openboe-menu" };
 	std::vector<unsigned int> spell_menus_connection_ids;
 
 	tgui::MenuBar::Ptr build_menubar() const;

--- a/src/pcedit/pc.menu.hpp
+++ b/src/pcedit/pc.menu.hpp
@@ -17,11 +17,11 @@ public:
 	
 private:
 	
-	using MenuHierarchy = std::vector<sf::String>;
+	using MenuHierarchy = std::vector<tgui::String>;
 	
 	tgui::Gui tgui;
 	sf::RenderWindow& mainPtr;
-	const sf::String internal_menubar_widget_name { "openboe-pcedit-menu" };
+	const tgui::String internal_menubar_widget_name { "openboe-pcedit-menu" };
 	
 	tgui::MenuBar::Ptr build_menubar() const;
 	void add_menu_placeholders(tgui::MenuBar::Ptr&) const;

--- a/src/scenedit/scen.menu.hpp
+++ b/src/scenedit/scen.menu.hpp
@@ -22,10 +22,10 @@ public:
 	
 private:
 	
-	using MenuHierarchy = std::vector<sf::String>;
+	using MenuHierarchy = std::vector<tgui::String>;
 	
 	tgui::Gui tgui;
-	const sf::String internal_menubar_widget_name { "openboe-scenedit-menu" };
+	const tgui::String internal_menubar_widget_name { "openboe-scenedit-menu" };
 
 	tgui::MenuBar::Ptr build_menubar() const;
 	void add_menu_placeholders(tgui::MenuBar::Ptr&) const;

--- a/src/tools/winutil.hpp
+++ b/src/tools/winutil.hpp
@@ -12,6 +12,7 @@
 #include <boost/filesystem/path.hpp>
 #include <SFML/Window.hpp>
 #include <SFML/Graphics/Image.hpp>
+#include <memory>
 
 char keyToChar(sf::Keyboard::Key key, bool isShift);
 

